src/App.vue

```diff
@@ -1,28 +1,131 @@
+                   :key="timezoneValue"
+                   class="timezone-card-wrapper">
+          <timezone-card :timezone-name="availableTimezones.get(timezoneValue).text"
+                         :timezone-key="availableTimezones.get(timezoneValue).utc[0]"
+                         v-on:remove="removeTimezone(timezoneValue)"
+                         class="md-flex">
+          </timezone-card>
+        </md-layout>
+      </md-layout>
+    </div>
+  
+    <clear-all-confirmation-dialog ref="clear-all-confirmation-dialog"
+                                   v-on:confirm="clearAll();">
+    </clear-all-confirmation-dialog>
+  
+    <add-timezone-dialog ref="add-timezone-dialog"
+                         :timezones="notAddedTimezones"
+                         v-on:add-timezone="addTimezone">
+    </add-timezone-dialog>
   </div>
 </template>
 
 <script>
-import Hello from './components/Hello'
+import TimezoneCard from './components/TimezoneCard'
+import ClearAllConfirmationDialog from './components/ClearAllConfirmationDialog'
+import AddTimezoneDialog from './components/AddTimezoneDialog'
 
 export default {
   name: 'app',
   components: {
-    Hello
+    TimezoneCard, ClearAllConfirmationDialog, AddTimezoneDialog
+  },
+  props: ['timezones'],
+  data: function () {
+    return {
+      availableTimezones: new Map(),
+      shownTimezones: []
+    }
+  },
+  created: function () {
+    // Add timezones from allTimezones into our available timezones Map
+    for (let i = 0; i < this.timezones.length; ++i) {
+      const timezone = this.timezones[i]
+      if (timezone.utc && timezone.utc.length > 0) {
+        // Omit timezones withou utc field, the utc field is used to identify
+        // the timezone in momenttz.js to get the proper current time.
+        // Timezones withou utc field are old and unused (nothing returned by googling of such timezones)
+        // At the time of writing (10.4.2017) there were 2 such timezones
+        this.availableTimezones.set(timezone.value, timezone)
+      }
+    }
+
+    // Try to load previously shown timezones from localStorage
+    if (typeof (Storage) !== 'undefined' && localStorage.shownTimezones) {
+      // Browser supports localStorage and shownTimezones are stored
+      // Make sure all stored timezones are contained in available timezones, otherwise drop them
+      const storedTimezones = JSON.parse(localStorage.shownTimezones)
+      for (let i = 0; i < storedTimezones.length; ++i) {
+        const storedTimezoneValue = storedTimezones[i]
+        if (this.availableTimezones.has(storedTimezoneValue)) {
+          this.shownTimezones.push(storedTimezoneValue)
+        }
+      }
+    }
+  },
+  computed: {
+    availableTimezonesList() {
+      return Array.from(this.availableTimezones.values())
+        .sort((left, right) => left.offset - right.offset)
+    },
+    notAddedTimezones() {
+      return this.availableTimezonesList.filter(e => this.shownTimezones.indexOf(e.value) === -1)
```

Comment:
I believe you may use there this.shownTimezones.includes() which returns a boolean